Plutarch/Lift.hs

```diff
@@ -64,30 +67,31 @@ pconstant x = punsafeConstantInternal $ PLC.someValue @(PConstantRepr (PLifted p
 
 -- | Error during script evaluation.
 data LiftError
-  = LiftError_ScriptError Scripts.ScriptError
-  | LiftError_EvalException (ErrorWithCause (MachineError PLC.DefaultFun) ())
+  = LiftError_EvalError EvalError
+  | LiftError_ReadKnownError (ErrorWithCause ReadKnownError (MachineError PLC.DefaultFun))
   | LiftError_FromRepr
-  | LiftError_WrongRepr
-  deriving stock (Eq, Show)
+  deriving stock (Eq)
 
 {- | Convert a Plutarch term to the associated Haskell value. Fail otherwise.
 This will fully evaluate the arbitrary closed expression, and convert the resulting value.
 -}
 plift' :: forall p. PUnsafeLiftDecl p => ClosedTerm p -> Either LiftError (PLifted p)
-plift' prog = case evaluateScript (compile prog) of
-  Right (_, _, Scripts.unScript -> UPLC.Program _ _ term) ->
+plift' prog = case evalScript (compile prog) of
+  (Right (Scripts.unScript -> UPLC.Program _ _ term), _, _) ->
     case readKnownConstant @_ @(PConstantRepr (PLifted p)) @(MachineError PLC.DefaultFun) Nothing term of
       Right r -> case pconstantFromRepr r of
         Just h -> Right h
         Nothing -> Left LiftError_FromRepr
-      Left e -> Left $ LiftError_EvalException e
-  Left e -> Left $ LiftError_ScriptError e
+      Left e -> Left $ LiftError_ReadKnownError e
+  (Left e, _, _) -> Left $ LiftError_EvalError e
 
 -- | Like `plift'` but throws on failure.
 plift :: forall p. (HasCallStack, PLift p) => ClosedTerm p -> PLifted p
 plift prog = case plift' prog of
   Right x -> x
-  Left e -> error $ "plift failed: " <> show e
+  Left LiftError_FromRepr -> error "plift failed because of pconstantFromRepr"
+  Left (LiftError_ReadKnownError _) -> error "plift failed because of an internal error in plutus-core"
```

Comment:
Good question, no idea.

Reply:
I guess stringifying `ReadKnownError` might only be possible with prisms
<https://playground.plutus.iohkdev.io/doc/haddock/plutus-core/html/PlutusCore-Evaluation-Machine-Exception.html#v:_UnliftingError>
<https://playground.plutus.iohkdev.io/doc/haddock/plutus-core/html/PlutusCore-Evaluation-Machine-Exception.html#v:_UnliftingErrorE>

that's really unfortunate. Best we can do to make this more informative might be:
```hs
Left (LiftError_ReadKnownError (ErrorWithCause _ errTerm)) ->
  error $ "plift failed because of an internal error in plutus-core"
    <> "\n"
    <> maybe "" (\x -> "cause: " <> show x) errTerm
```

Reply:
#319